egui/src/memory.rs

```diff
@@ -373,6 +374,44 @@ impl Memory {
     }
 }
 
+impl Memory {
+    pub fn get<T: DataElementTrait>(&mut self, id: Id) -> Option<&T> {
+        self.data.get_mut(&id)?.get()
+    }
+
+    pub fn get_or_insert<T: DataElementTrait>(&mut self, id: Id, or_insert: T) -> Option<&T> {
+        self.data
+            .entry(id)
+            .or_insert_with(|| DataElement::new(or_insert))
+            .get()
+    }
+
+    pub fn set<T: DataElementTrait>(&mut self, id: Id, set: T) {
+        self.data.insert(id, DataElement::new(set));
+    }
+
+    // Not all can be resetted, if it's not deserialized
+    pub fn reset_all<T: DataElementTrait>(&mut self) {
+        self.data
+            .retain(|_, v| (*v).type_id() == Some(std::any::TypeId::of::<T>()));
+    }
+
+    // Not anything can be counted
+    pub fn count<T: DataElementTrait>(&mut self) -> usize {
```

Comment:
Right now counting is only used for some inspection views, and I struggle to find any real-life usage of it.

Reply:
That's clever! With this approach, we can really count things and reset individual types. Unfortunately, `TypeId` is not de/serializable, so I introduced `SerializableTypeId` using, which has drawbacks because implemented using `Hash`.

Reply:
It is a shame that `TypeId` cannot be easily serialized. In [the documentation](https://doc.rust-lang.org/std/any/struct.TypeId.html) they make it clear that "the hashes will vary between Rust releases". This means upgrading the rust compiler may void the egui memory. As stated before the memory shouldn't be used for anything important, but I wonder if we can do better.

For instance, we could serialize the hash of the TypeId with each `DataElement` value. Then we could use that for counting and resetting, and if the hash breaks between releases only counting and resetting will break, but the memory will still be intact.

Reply:
Did that in next commits, also write what could break due to this in the documentation.